testcases/kernel/syscalls/recvmmsg/recvmmsg01.c

```diff
@@ -0,0 +1,2389 @@
+ *   each other and the tests, unsupectingly, pass the test vector via the
+ *   con_call_vec and con_call_n members.  Note that con_call_n == 1 is used
+ *   as a degenerate case to test recvmmsg(2) with a single message. Note
+ *   when recvmmsg(2) is used the messages can be sent individually or in
+ *   a single sendmmsg(2) for testing multi-message sending. Both ways of
+ *   sending are done controlled by con_call_sendmmsg == true to indicate
+ *   used of sendmmsg(2), otherwise multiple sendmsg(2) are used.
+ */
+
+typedef struct {
+	int		  con_client;
+	int		  con_server;
+	int		  con_type;
+	sockaddr_in_t	  con_server_sin;
+	bool		  con_call_sendmmsg;
+	bool		  con_sender_sleeps;
+	bool		  con_waitforone;
+	bool		  con_timeout;
+	size_t		  con_call_end;	/* index of last+1 filled */
+	size_t		  con_call_n;
+	call_t		**con_call_vec;	/* points to array of con_call_n */
+					/* call_t pointers to be used in */
+					/* a recvmmsg(2) multi-message test */
+	sendmsg_call_t	**con_smc_vec;
+	recvmsg_call_t	**con_rmc_vec;
+} con_t;
+
+typedef int (*calltest_t)(call_t *callp, con_t *conp, int tn);
+typedef void (*callconinit_t)(call_t *callp, con_t *conp);
+
+struct call_s {
+	calltest_t	call_test;
+	callconinit_t	call_con_init;
+	int		call_errno;	/* set on first run */
+};
+
+typedef struct {
+	call_t	call_base;		/* must be first: inherits call_t */
+	size_t	call_niov;
+} call_receive_iovec_boundary_checks_t;
+
+typedef struct {
+	call_t	call_base;		/* must be first: inherits call_t */
+	size_t	call_niov;
+} call_receive_iovec_boundary_checks_peeking_t;
+
+typedef struct {
+	call_t	call_base;		/* must be first: inherits call_t */
+	int	call_cloexec_flag;
+	bool	call_some_data;
+	ssize_t	call_controllen_delta;
+} call_receive_file_descriptor_t;
+
+typedef struct {
+	call_t	call_base;		/* must be first: inherits call_t */
+	int	call_type;
+} call_message_too_long_to_fit_might_discard_bytes_t;
+
+typedef struct {
+	call_t	call_base;		/* must be first: inherits call_t */
+} call_receive_waits_for_message_t;
+
+typedef struct {
+	call_t	call_base;		/* must be first: inherits call_t */
+} call_receiver_doesnt_wait_for_messages_t;
+
+typedef struct {
+	call_t	call_base;		/* must be first: inherits call_t */
+} call_receive_returns_what_is_available_t;
+
+typedef struct {
+	call_t	call_base;		/* must be first: inherits call_t */
+} call_empty_datagram_received_as_empty_datagram_t;
+
+/*}
+ *{	Prototypes.
+ *
+ *	These functions are all static because that is what the LTP coding
+ *	convention likes, doing this makes no different other than making
+ *	the code harder to debug. To be able to make them non-static for
+ *	easier debugging (for example so their names are in the symbol table
+ *	for core dump debugging, etc) STATIC is used instead, which can be
+ *	easily changed to remove the static declarator which gets in the way
+ *	of putting symbols in the symbol table.
+ */
+
+// #define STATIC
+#define	STATIC static
```

Comment:
Uh, why do you even do this?

And it's even commented with // #define STATIC comment, this is beyond commenting obvious.